frontend/src/app/testQueue/QueueItem.tsx

```diff
@@ -316,11 +312,8 @@ const QueueItem: any = ({
     showNotification(toast, alert);
   };
 
-  const onTestResultSubmit = (e?: React.MouseEvent<HTMLButtonElement>) => {
-    setIsSubmitting(true);
```

Comment:
I think this was probably the real issue: the `isSubmitting` state variable is redundant with `loading` from `useMutation` and is clearly set to `true` too early here.